server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java

```diff
@@ -1000,4 +1020,97 @@ public void testAllFeatures() {
 
         assertEquals(0, remaining);
     }
+
+    private static Set<SnapshotId> removeRandomSnapshotDetails(List<String> repositories) {
+        if (randomBoolean()) {
+            return Set.of();
+        }
+        final var masterRepositoriesService = internalCluster().getCurrentMasterNodeInstance(RepositoriesService.class);
+        return safeAwait(l0 -> {
+            final Set<SnapshotId> snapshotsWithoutDetails = ConcurrentCollections.newConcurrentSet();
+            try (var listeners = new RefCountingListener(l0.map(v -> Set.copyOf(snapshotsWithoutDetails)))) {
+                for (final var repositoryName : repositories) {
+                    if (randomBoolean()) {
+                        continue;
+                    }
+
+                    final var repository = asInstanceOf(FsRepository.class, masterRepositoriesService.repository(repositoryName));
+                    final var repositoryMetadata = repository.getMetadata();
+                    final var repositorySettings = repositoryMetadata.settings();
+                    final var rootPath = asInstanceOf(FsBlobStore.class, repository.blobStore()).path();
+
+                    SubscribableListener
+
+                        .<AcknowledgedResponse>newForked(
+                            l -> client().execute(
+                                TransportDeleteRepositoryAction.TYPE,
+                                new DeleteRepositoryRequest(TEST_REQUEST_TIMEOUT, TEST_REQUEST_TIMEOUT, repositoryName),
+                                l
+                            )
+                        )
+                        .andThenAccept(ElasticsearchAssertions::assertAcked)
+                        .andThenAccept(v -> {
+                            final var repoDataBlobPath = rootPath.resolve(
+                                BlobStoreRepository.INDEX_FILE_PREFIX + repositoryMetadata.generation()
+                            );
+                            final var repoDataBytes = Files.readAllBytes(repoDataBlobPath);
+                            final var repoDataMap = XContentHelper.convertToMap(
+                                JsonXContent.jsonXContent,
+                                repoDataBytes,
+                                0,
+                                repoDataBytes.length,
+                                true
+                            );
+                            final var snapshotsList = asInstanceOf(List.class, repoDataMap.get("snapshots"));
```

Comment:
```suggestion
                           /**
                             * Here are the actual metadata changes. We will randomly select snapshots from which to remove fields and
                             * update the blob store with the modified metadata. The fields being removed did not exist in previous versions
                             * of elasticsearch and the latest version must maintain backwards compatibility.
                             */

                            final var snapshotsList = asInstanceOf(List.class, repoDataMap.get("snapshots"));
```

Reply:
There's a lot of code in the method, I'd like a comment pointer to the important bits to help zero in for understanding.